src/components/backgroundselector/BackgroundService.js

```diff
@@ -145,18 +145,22 @@ goog.require('ga_glstylestorage_service');
         });
       };
 
-      var createOlLayer = function(map, bg) {
-        var layer = bg.olLayer;
-        if (!layer) {
-          layer = gaLayers.getOlLayerById(bg.id, {
-            externalStyleUrl: bg.externalStyleUrl
-          });
-          layer.adminId = bg.adminId;
-          layer.background = true;
-          layer.displayInLayerManager = false;
-          bg.olLayer = layer;
+      var createOlLayer = function(bg) {
+        var externalStyleUrl;
+        if (bg.externalStyleUrl) {
+          externalStyleUrl = bg.externalStyleUrl;
+        } else {
+          externalStyleUrl = bg.olLayer ?
+            bg.olLayer.externalStyleUrl : undefined;
         }
-        return layer;
+        bg.olLayer = gaLayers.getOlLayerById(bg.id, {
```

Comment:
Why we need to recreate the layer eachtime ?

Reply:
because of the ol bug, I guess

Reply:
We never re-used layers for a reason. I think we can prevent a lot of unwanted bugs by simply re-creating a new instance each time. This stands for future versions of ol as well.